extensions/arm/GL_ARM_tensors.txt

```diff
@@ -0,0 +1,318 @@
+        void tensorReadARM(tensorARM t, uint coords[], out uint16_t data, uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], out uint32_t data, uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], out uint64_t data, uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], out float16_t data, uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], out float32_t data, uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], out float64_t data, uint tensorOperands = 0U, ...)
+
+        void tensorReadARM(tensorARM t, uint coords[], bool data[], uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], int8_t data[], uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], int16_t data[], uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], int32_t data[], uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], int64_t data[], uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], uint8_t data[], uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], uint16_t data[], uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], uint32_t data[], uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], uint64_t data[], uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], float16_t data[], uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], float32_t data[], uint tensorOperands = 0U, ...)
+        void tensorReadARM(tensorARM t, uint coords[], float64_t data[], uint tensorOperands = 0U, ...)
+
+    Description: Read one, or more, elements from Tensor *t*. The number of
+    elements read is specified by the type of the *data* parameter.
+    If *data* is a scalar type, a single element is read.
+    If *data* is an array, the number of elements read equals the array size.
+    The type of element(s) read is given by the tensorARM type.
+    *coords* specifies the coordinates of the first element to read.
+    The first element of the *coords* array corresponds to the outermost
+    dimension of the tensor.
+    It is a compile-time error if the number of supplied coordinates does not
+    equal the rank of the tensor.
+    *tensorOperands* is a constant bitmask of zero or more gl_TensorOperands*
+    values.
+    Depending on the specified tensorOperands, additional arguments may have to
+    be specified.
+    The behavior when one or more tensor elements read are outside of the tensor
+    bounds and gl_TensorOperandsOutOfBoundsValueARM is not provided, is specified
+    by the client API.
```

Comment:
If you're reading more than 1 element do they all have to be contained within a single dimension, or can the coordinates wrap around in some way? I could imagine that if you read to the end of one row maybe it would carry on in the next.